File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Configuration/Conventions/Caching.cs
namespace Raven.Documentation.Samples.ClientApi.Configuration.Conventions
{
	using Client.Document;

	public class Caching
	{
		public Caching()
		{
			var store = new DocumentStore();

			DocumentConvention Conventions = store.Conventions;

			#region should_cache
			Conventions.ShouldCacheRequest = url => true;
			#endregion

			#region should_aggressive_cache_track_changes
			Conventions.ShouldAggressiveCacheTrackChanges = true;
			#endregion

			#region should_save_changes_force_aggressive_cache_check
			Conventions.ShouldSaveChangesForceAggressiveCacheCheck = true;
			#endregion
		} 
	}
}

File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/configuration/conventions/Caching.java
package net.ravendb.clientapi.configuration.conventions;

import net.ravendb.client.delegates.RequestCachePolicy;
import net.ravendb.client.document.DocumentConvention;
import net.ravendb.client.document.DocumentStore;


public class Caching {

  public Caching() {
    DocumentStore store = new DocumentStore();
    DocumentConvention conventions = store.getConventions();

    //region should_cache
    conventions.setShouldCacheRequest(new RequestCachePolicy() {
      @Override
      public Boolean shouldCacheRequest(String url) {
        return Boolean.TRUE;
      }
    });
    //endregion

    //region should_aggressive_cache_track_changes
    conventions.setShouldAggressiveCacheTrackChanges(true);
    //endregion

    //region should_save_changes_force_aggressive_cache_check
    conventions.setShouldSaveChangesForceAggressiveCacheCheck(true);
    //endregion
  }
}
